ecosystem/sep-0043.md

```diff
@@ -0,0 +1,215 @@
+    } | Error>,
+
+  signMessage: (
+    message: string,
+    opts?: {
+      networkPassphrase?: string,
+      address?: string
+    }) =>
+    Promise<{
+      signedMessage: string;
+      signerAddress: string;
+    } | Error>,
+
+  getNetwork: () =>
+    Promise<{
+      network: string,
+      networkPassphrase: string
+    }>
+}
+```
+
+In addition, this proposal has a standard for error interfaces.
+
+The errors will return a specific code depending on the type of error that the wallet is encountering. Codes map as
+follows:
+
+```
+-1: internal wallet error, likely caused by the wallet logic itself
+-2: external service (Horizon, RPC, etc.) returned an error
+-3: client app request is invalid (wrong parameters, invalid transaction XDR, etc.)
+-4: user rejected the request, client app should not try to retry the request
+```
+
+Alongside these error codes, the wallet will include a `message` key that maps to a custom error string that the wallet
+can define. This string will be a human readable message that informs the user of what went wrong.
+
+Wallets may also include an optional `ext` field which is an array of strings that may provide some specific information
+about the error.
+
+Below are examples of different kinds of errors a wallet may surface:
+
+```typescript
+  // an internal error occurred in the wallet. For example, a JS runtime error
+
+  {
+    message: 'The wallet encountered an internal error. Please try again or contact the wallet if the problem persists.',
+    code: -1
+  },
+
+  // an error occurred with an external service. For example, Horizon returned an error
+
+  {
+    message: 'An error occurred with Horizon. Please try again.',
+    code: -2,
+    ext: ['Operation 2: Invalid "amount" value.', "Operation 5: Asset issuer is required."] // malformed tx error example
+  },
+
+  // Either the dapp or the user passed a malformed request to the wallet. For example, a user didn't pass a valid XDR to `signTransaction`
+
+  {
+    message: 'Request is invalid. Please check the details and try again.',
+    code: -3,
+    ext: ['Invalid transaction XDR'] // example error
+  },
+
+  // The wallet received the request from the client, but the user decided to decline.
+
+  rejected: {
+    message: 'The user rejected this request.',
+    code: -4
+  }
+
+```
+
+### Methods
+
+### getAddress
+
+`getAddress` will provide the public key (G...) the wallet is signing for. This may be the account loaded in the wallet,
+or it may be a different account that the wallet is the signer for.
+
+### signTransaction
+
+`signTransaction` will take an XDR provided by the requester and apply a signature to it before returning a
+base64-encoded string XDR-encoded
+[Transaction Envelope](https://github.com/stellar/js-stellar-base/blob/5efdff0b2a78ed842f7463340b4ff23eb4370635/types/next.d.ts#L11928)
+with
+[Decorated Signatures](https://github.com/stellar/js-stellar-base/blob/5efdff0b2a78ed842f7463340b4ff23eb4370635/types/next.d.ts#L5882)
```

Comment:
```suggestion
[Decorated Signatures](https://github.com/stellar/js-stellar-base/blob/5efdff0b2a78ed842f7463340b4ff23eb4370635/types/curr.d.ts#L5882)
```

(They're identical today, but referencing `curr` will be more accurate in the future once `next` changes.)